app/src/main/java/io/github/iurimenin/upcomingmovies/presenter/MoviesPresenter.kt

```diff
@@ -0,0 +1,75 @@
+package io.github.iurimenin.upcomingmovies.presenter
+
+import android.content.Context
+import android.content.Intent
+import android.support.design.widget.Snackbar
+import android.support.v4.app.Fragment
+import android.view.View
+import android.widget.ProgressBar
+import io.github.iurimenin.upcomingmovies.R
+import io.github.iurimenin.upcomingmovies.model.MovieVO
+import io.github.iurimenin.upcomingmovies.util.Utils
+import io.github.iurimenin.upcomingmovies.view.MovieAdapter
+import io.github.iurimenin.upcomingmovies.view.MovieDetailActivity
+
+/**
+ * Created by Iuri Menin on 26/08/17.
+ */
+class MoviesPresenter : AsyncTaskCallback {
+
+    private var mContext: Context? = null
+    private var mMovieAdapter: MovieAdapter? = null
+    private var mMoviesProgressBar : ProgressBar? = null
+    private var mView: View? = null
```

Comment:
The presenter class is referencing Android widgets. Is it on purpose?

Reply:
Yes, I have a reference of the ProgressBar to change his visibility, but I think this isn't the best option, right?